code/modules/reagents/chemistry/reagents/Chemistry-Reagents-Core.dm

```diff
@@ -79,6 +79,8 @@
 		M.heal_organ_damage(0.2 * removed * volume_mod, 0)	// Heal brute slightly like normal nutrition. More 'effective' blood means more usable material.
 		M.adjust_hydration(2 * removed) // Still has some water in the form of plasma. Hydrates less than a normal drink.
 		M.add_chemical_effect(CE_BLOODRESTORE, 4 * removed) //same rating as eating nutriment
+		if(effective_dose >= 20)
```

Comment:
```suggestion
		if(effective_dose >= 20 && prob(20))
```
Without any 'prob' checks here, the mob is likely gonna end up vomiting violently and heavily. Vomit is a pretty powerful stun. Punish people if you'd like for drinking too much, but add a chance or something if you do this.

Vomiting doesn't clear any reagents that are ingested either.